### PR TITLE
Check hash of firmware

### PR DESCRIPTION
**Pull request by qistoph**
This PR sends an MD5 hash along with the uploaded firmware and checks on the ESP if the received data matches the hash.

It's based on the same functionality in [ESP8266httpUpdate](https://github.com/esp8266/Arduino/tree/master/libraries/ESP8266httpUpdate).

I have not updated the server side code in this repo yet, but a working example is included in the [PR at your AsyncElegantOTA repo](https://github.com/ayushsharma82/AsyncElegantOTA/pull/29).

TODO:
- [ ] Handle MD5 param in ESP8266 code
- [ ] Handle MD5 param in ESP32 code

I would like your feedback on this to determine if it's worth spending more time on the implementation for the (Sync)ElegantOTA server side.

### CODE REVIEW

**Comment by ayushsharma82** [comment]
Interesting! We don't use the `Update` class directly in this library ( unlike AsyncElegantOTA ), Therefore it would be better not to spend any time rewriting the core routines from the ground up.

Let's keep the MD5 check for the AsyncElegantOTA for the time being. It's a nice addition though.

**Comment by qistoph** [comment]
I'll try to work out a PR for the Update class to see if they're willing to include (optional) support for MD5 there 🙂

File: ui/src/App.vue [hunk 1]
```diff
@@ -187,12 +234,8 @@ export default {
   },
 
   mounted() {
-    fetch('/update/identity').then(async (response) => {
```
Comment (ayushsharma82) [inline ui/src/App.vue:190]: I think you made this to test in the local environment. 
Make sure to revert these lines for deployment on an MCU.

Comment (ayushsharma82) [reply ui/src/App.vue:190]: We can use `if(process.env.NODE_ENV === 'production')` to skip this HTTP request

Comment (qistoph) [reply ui/src/App.vue:190]: You're right. I should've excluded that in the PR.

Comment (qistoph) [reply ui/src/App.vue]: Added a commit to correct his.

**Pull request merged**